<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Advanced: Comparison to reference dataset • dyngen</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Advanced: Comparison to reference dataset">
<meta property="og:description" content="dyngen">
<meta property="og:image" content="https://dyngen.dynverse.org/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-7YGMM4L0JM"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-7YGMM4L0JM');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">dyngen</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/installation.html">Installation instructions</a>
    </li>
    <li>
      <a href="../../articles/getting_started.html">Getting started</a>
    </li>
    <li>
      <a href="../../articles/showcase_backbones.html">Showcase different backbones</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced</li>
    <li>
      <a href="../../articles/advanced_topics/comparison_reference.html">Advanced: Comparison to reference dataset</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/constructing_backbone.html">Advanced: Constructing a custom backbone</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/run_dyngen_from_docker.html">Advanced: Running dyngen from a docker container</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/scalability_and_runtime.html">Advanced: On scalability and runtime</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/simulating_batch_effects.html">Advanced: Simulating batch effects</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/simulating_knockouts.html">Advanced: Simulating a knockout experiment</a>
    </li>
    <li>
      <a href="../../articles/advanced_topics/tweaking_parameters.html">Advanced: Tweaking parameters</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dynverse/dyngen/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="comparison_reference_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Advanced: Comparison to reference dataset</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/dynverse/dyngen/blob/master/vignettes/advanced_topics/comparison_reference.Rmd"><code>vignettes/advanced_topics/comparison_reference.Rmd</code></a></small>
      <div class="hidden name"><code>comparison_reference.Rmd</code></div>

    </div>

    
    
<!-- github markdown built using 
rmarkdown::render("vignettes/advanced_topics/comparison_reference.Rmd", output_format = rmarkdown::github_document(html_preview = FALSE))
-->
<p>In this vignette, we will take a look at characteristic features of dyngen versus the reference dataset it uses. To this end, we’ll be using <a href="https://www.bioconductor.org/packages/release/bioc/html/countsimQC.html"><code>countsimQC</code></a> <span class="citation">(Soneson and Robinson 2018)</span> to calculate key statistics of both datasets and create comparative visualisations.</p>
<div id="run-dyngen-simulation" class="section level2">
<h2 class="hasAnchor">
<a href="#run-dyngen-simulation" class="anchor"></a>Run dyngen simulation</h2>
<p>We use an internal function from the dyngen package to download and cache one of the reference datasets.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dyngen.dynverse.org">dyngen</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"realcounts"</span>, package <span class="op">=</span> <span class="st">"dyngen"</span><span class="op">)</span>
<span class="va">name_realcounts</span> <span class="op">&lt;-</span> <span class="st">"zenodo_1443566_real_silver_bone-marrow-mesenchyme-erythrocyte-differentiation_mca"</span>
<span class="va">url_realcounts</span> <span class="op">&lt;-</span> <span class="va">realcounts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="va">name_realcounts</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">url</span><span class="op">)</span>
<span class="va">realcount</span> <span class="op">&lt;-</span> <span class="fu">dyngen</span><span class="fu">:::</span><span class="fu">.download_cacheable_file</span><span class="op">(</span><span class="va">url_realcounts</span>, <span class="fu"><a href="https://rdrr.io/r/base/options.html">getOption</a></span><span class="op">(</span><span class="st">"dyngen_download_cache_dir"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>We run a simple dyngen dataset as follows, where the number of cells and genes are determined by the size of the reference dataset.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">backbone</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/list_backbones.html">backbone_bifurcating_loop</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">num_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">realcount</span><span class="op">)</span>
<span class="va">num_feats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">realcount</span><span class="op">)</span>
<span class="va">num_tfs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">backbone</span><span class="op">$</span><span class="va">module_info</span><span class="op">)</span>
<span class="va">num_tar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="op">(</span><span class="va">num_feats</span> <span class="op">-</span> <span class="va">num_tfs</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">num_hks</span> <span class="op">&lt;-</span> <span class="va">num_feats</span> <span class="op">-</span> <span class="va">num_tfs</span> <span class="op">-</span> <span class="va">num_tar</span>

<span class="va">config</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../../reference/initialise_model.html">initialise_model</a></span><span class="op">(</span>
    backbone <span class="op">=</span> <span class="va">backbone</span>,
    num_cells <span class="op">=</span> <span class="va">num_cells</span>,
    num_tfs <span class="op">=</span> <span class="va">num_tfs</span>,
    num_targets <span class="op">=</span> <span class="va">num_tar</span>,
    num_hks <span class="op">=</span> <span class="va">num_hks</span>,
    gold_standard_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_gold_standard.html">gold_standard_default</a></span><span class="op">(</span><span class="op">)</span>,
    simulation_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_cells.html">simulation_default</a></span><span class="op">(</span>
      total_time <span class="op">=</span> <span class="fl">1000</span>,
      experiment_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_cells.html">simulation_type_wild_type</a></span><span class="op">(</span>num_simulations <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
    <span class="op">)</span>,
    experiment_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_experiment.html">experiment_snapshot</a></span><span class="op">(</span>
      realcount <span class="op">=</span> <span class="va">realcount</span>
    <span class="op">)</span>,
    verbose <span class="op">=</span> <span class="cn">FALSE</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the simulation is being sped up because rendering all vignettes with one core</span>
<span class="co"># for pkgdown can otherwise take a very long time</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>

<span class="va">config</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../../reference/initialise_model.html">initialise_model</a></span><span class="op">(</span>
    backbone <span class="op">=</span> <span class="va">backbone</span>,
    num_cells <span class="op">=</span> <span class="va">num_cells</span>,
    num_tfs <span class="op">=</span> <span class="va">num_tfs</span>,
    num_targets <span class="op">=</span> <span class="va">num_tar</span>,
    num_hks <span class="op">=</span> <span class="va">num_hks</span>,
    verbose <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html">interactive</a></span><span class="op">(</span><span class="op">)</span>,
    download_cache_dir <span class="op">=</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html">R_user_dir</a></span><span class="op">(</span><span class="st">"dyngen"</span>, <span class="st">"data"</span><span class="op">)</span>,
    simulation_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_cells.html">simulation_default</a></span><span class="op">(</span>
      total_time <span class="op">=</span> <span class="fl">1000</span>,
      census_interval <span class="op">=</span> <span class="fl">2</span>, 
      ssa_algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GillespieSSA2/man/ssa_etl.html">ssa_etl</a></span><span class="op">(</span>tau <span class="op">=</span> <span class="fl">300</span><span class="op">/</span><span class="fl">3600</span><span class="op">)</span>,
      experiment_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_cells.html">simulation_type_wild_type</a></span><span class="op">(</span>num_simulations <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
    <span class="op">)</span>,
    experiment_params <span class="op">=</span> <span class="fu"><a href="../../reference/generate_experiment.html">experiment_snapshot</a></span><span class="op">(</span>
      realcount <span class="op">=</span> <span class="va">realcount</span>
    <span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/generate_dataset.html">generate_dataset</a></span><span class="op">(</span><span class="va">config</span>, make_plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/dyngen_generate-1.png" width="1152"></p>
<p>Both datasets are stored in a list for easy usage by countsimQC.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">datasets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  real <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">realcount</span><span class="op">)</span><span class="op">)</span>,
  dyngen <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">ddsList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">datasets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">ds</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>
    countData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">ds</span><span class="op">)</span><span class="op">)</span>, 
    colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">ds</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, 
    design <span class="op">=</span> <span class="op">~</span><span class="fl">1</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="run-countsimqc-computations" class="section level2">
<h2 class="hasAnchor">
<a href="#run-countsimqc-computations" class="anchor"></a>Run countsimQC computations</h2>
<p>Below are some computations countsimQC makes. Normally these are not visible to the user, but for the sake of transparency these are included in the vignette.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/csoneson/countsimQC">countsimQC</a></span><span class="op">)</span>

<span class="co">## Define helper objects</span>
<span class="va">nDatasets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ddsList</span><span class="op">)</span>
<span class="va">colRow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>
<span class="va">panelSize</span> <span class="op">&lt;-</span> <span class="fl">4</span>
<span class="va">thm</span> <span class="op">&lt;-</span> 
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">theme</span><span class="op">(</span>
    axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,
    axis.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,
    strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<p>Compute key characteristics</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu">countsimQC</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/countsimQC/man/calculateDispersionsddsList.html">calculateDispersionsddsList</a></span><span class="op">(</span>ddsList <span class="op">=</span> <span class="va">ddsList</span>, maxNForDisp <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>

<span class="va">sampleCorrDF</span> <span class="op">&lt;-</span> <span class="fu">countsimQC</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/countsimQC/man/calculateSampleCorrs.html">calculateSampleCorrs</a></span><span class="op">(</span>ddsList <span class="op">=</span> <span class="va">obj</span>, maxNForCorr <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>

<span class="va">featureCorrDF</span> <span class="op">&lt;-</span> <span class="fu">countsimQC</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/countsimQC/man/calculateFeatureCorrs.html">calculateFeatureCorrs</a></span><span class="op">(</span>ddsList <span class="op">=</span> <span class="va">obj</span>, maxNForCorr <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<p>Summarize sample characteristics</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleDF</span> <span class="op">&lt;-</span> <span class="fu">map2_df</span><span class="op">(</span><span class="va">obj</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">dataset_name</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
    dataset <span class="op">=</span> <span class="va">dataset_name</span>,
    Libsize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
    Fraczero <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>,
    TMM <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">norm.factors</span>,
    EffLibsize <span class="op">=</span> <span class="va">Libsize</span> <span class="op">*</span> <span class="va">TMM</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>Summarize feature characteristics</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">featureDF</span> <span class="op">&lt;-</span> <span class="fu">map2_df</span><span class="op">(</span><span class="va">obj</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">dataset_name</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">rd</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dds</span><span class="op">)</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
    dataset <span class="op">=</span> <span class="va">dataset_name</span>,
    Tagwise <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">tagwise.dispersion</span><span class="op">)</span>,
    Common <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">common.dispersion</span><span class="op">)</span>,
    Trend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">trended.dispersion</span><span class="op">)</span>,
    AveLogCPM <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">AveLogCPM</span>,
    AveLogCPMDisp <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">AveLogCPMDisp</span>, 
    average_log2_cpm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span>, prior.count <span class="op">=</span> <span class="fl">2</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">mean</span><span class="op">)</span>, 
    variance_log2_cpm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span>, prior.count <span class="op">=</span> <span class="fl">2</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">var</span><span class="op">)</span>,
    Fraczero <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>,
    dispGeneEst <span class="op">=</span> <span class="va">rd</span><span class="op">$</span><span class="va">dispGeneEst</span>,
    dispFit <span class="op">=</span> <span class="va">rd</span><span class="op">$</span><span class="va">dispFit</span>,
    dispFinal <span class="op">=</span> <span class="va">rd</span><span class="op">$</span><span class="va">dispersion</span>,
    baseMeanDisp <span class="op">=</span> <span class="va">rd</span><span class="op">$</span><span class="va">baseMeanDisp</span>,
    baseMean <span class="op">=</span> <span class="va">rd</span><span class="op">$</span><span class="va">baseMean</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>Summarize data set characteristics</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">datasetDF</span> <span class="op">&lt;-</span> <span class="fu">map2_df</span><span class="op">(</span><span class="va">obj</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">dataset_name</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
    dataset <span class="op">=</span> <span class="va">dataset_name</span>,
    prior_df <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"prior.df = "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">prior.df</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
    nVars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
    nSamples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">dge</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
    AveLogCPMDisp <span class="op">=</span> <span class="fl">0.8</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">featureDF</span><span class="op">$</span><span class="va">AveLogCPMDisp</span><span class="op">)</span>,
    Tagwise <span class="op">=</span> <span class="fl">0.9</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">featureDF</span><span class="op">$</span><span class="va">Tagwise</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="data-set-dimensions" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#data-set-dimensions" class="anchor"></a>Data set dimensions</h2>
<p>These bar plots show the number of samples (columns) and features (rows) in each data set.</p>
<p>Number of samples (columns)</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">datasetDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">dataset</span>, y <span class="op">=</span> <span class="va">nSamples</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Number of samples (columns)"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="va">thm</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/nSamples-1.png" width="960"></p>
<p>Number of features (rows)</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">datasetDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">dataset</span>, y <span class="op">=</span> <span class="va">nVars</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Number of features (rows)"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="va">thm</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/nVariables-1.png" width="960"></p>
</div>
<div id="dispersionbcv-plots" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#dispersionbcv-plots" class="anchor"></a>Dispersion/BCV plots</h2>
<p>Disperson/BCV plots show the association between the average abundance and the dispersion or “biological coefficient of variation” (sqrt(dispersion)), as calculated by <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html"><code>edgeR</code></a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> and <a href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html"><code>DESeq2</code></a> <span class="citation">(Love, Huber, and Anders 2014)</span>. In the <code>edgeR</code> plot, the estimate of the prior degrees of freedom is indicated.</p>
<div id="edger" class="section level3">
<h3 class="hasAnchor">
<a href="#edger" class="anchor"></a>edgeR</h3>
<p>The black dots represent the tagwise dispersion estimates, the red line the common dispersion and the blue curve represents the trended dispersion estimates. For further information about the dispersion estimation in <code>edgeR</code>, see <span class="citation">Chen, Lun, and Smyth (2014)</span>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">AveLogCPMDisp</span><span class="op">)</span>, 
       <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">AveLogCPMDisp</span>, y <span class="op">=</span> <span class="va">Tagwise</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.25</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">Trend</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"blue"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">Common</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">datasetDF</span>, <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">prior_df</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Average log CPM"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Biological coefficient of variation"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/BCVedgeR-1.png" width="768"></p>
</div>
<div id="deseq2" class="section level3">
<h3 class="hasAnchor">
<a href="#deseq2" class="anchor"></a>DESeq2</h3>
<p>The black dots are the gene-wise dispersion estimates, the red curve the fitted mean-dispersion relationship and the blue circles represent the final dispersion estimates.For further information about the dispersion estimation in <code>DESeq2</code>, see <span class="citation">Love, Huber, and Anders (2014)</span>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">baseMeanDisp</span><span class="op">)</span>, 
       <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">baseMeanDisp</span>, y <span class="op">=</span> <span class="va">dispGeneEst</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.25</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">dispFinal</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"lightblue"</span>, shape <span class="op">=</span> <span class="fl">21</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">dispFit</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Base mean"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Dispersion"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/dispersionDESeq2-1.png" width="768"></p>
</div>
</div>
<div id="mean-variance-plots" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#mean-variance-plots" class="anchor"></a>Mean-variance plots</h2>
<p>This scatter plot shows the relation between the empirical mean and variance of the features. The difference between these mean-variance plots and the mean-dispersion plots above is that the plots in this section do not take the information about the experimental design and sample grouping into account, but simply display the mean and variance of log2(CPM) estimates across all samples, calculated using the <code>cpm</code> function from <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html"><code>edgeR</code></a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>, with a prior count of 2.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">average_log2_cpm</span>, y <span class="op">=</span> <span class="va">variance_log2_cpm</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.75</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Mean of log2(CPM)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Variance of log2(CPM)"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/meanVarSepScatter-1.png" width="768"></p>
</div>
<div id="library-sizes" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#library-sizes" class="anchor"></a>Library sizes</h2>
<p>This plot shows a histogram of the total read count per sample, i.e., the column sums of the respective data matrices.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Libsize</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Library size"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/libsizeSepHist-1.png" width="768"></p>
</div>
<div id="tmm-normalization-factors" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#tmm-normalization-factors" class="anchor"></a>TMM normalization factors</h2>
<p>This plot shows a histogram of the TMM normalization factors <span class="citation">(Robinson and Oshlack 2010)</span>, intended to adjust for differences in RNA composition, as calculated by <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html"><code>edgeR</code></a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">TMM</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"TMM normalization factor"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/tmmSepHist-1.png" width="768"></p>
</div>
<div id="effective-library-sizes" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#effective-library-sizes" class="anchor"></a>Effective library sizes</h2>
<p>This plot shows a histogram of the “effective library sizes”, defined as the total count per sample multiplied by the corresponding TMM normalization factor.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">EffLibsize</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Effective library size"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/effLibsizeSepHist-1.png" width="768"></p>
</div>
<div id="expression-distributions-average-log-cpm" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#expression-distributions-average-log-cpm" class="anchor"></a>Expression distributions (average log CPM)</h2>
<p>This plot shows the distribution of average abundance values for the features. The abundances are log CPM values calculated by <code>edgeR</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">AveLogCPM</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Average log CPM"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/logCPMSepHist-1.png" width="768"></p>
</div>
<div id="fraction-zeros-per-sample" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#fraction-zeros-per-sample" class="anchor"></a>Fraction zeros per sample</h2>
<p>This plot shows the distribution of the fraction of zeros observed per sample (column) in the count matrices.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Fraczero</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Fraction zeros per sample"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/fraczeroSampleSepHist-1.png" width="768"></p>
</div>
<div id="fraction-zeros-per-feature" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#fraction-zeros-per-feature" class="anchor"></a>Fraction zeros per feature</h2>
<p>This plot illustrates the distribution of the fraction of zeros observed per feature (row) in the count matrices.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Fraczero</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Fraction zeros per feature"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/fraczeroFeatureSepHist-1.png" width="768"></p>
</div>
<div id="sample-sample-correlations" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#sample-sample-correlations" class="anchor"></a>Sample-sample correlations</h2>
<p>The plot below shows the distribution of Spearman correlation coefficients for pairs of samples, calculated from the log(CPM) values obtained via the <code>cpm</code> function from <code>edgeR</code>, with a prior.count of 2.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleCorrDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Sample-sample correlation"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/sampleCorrSepHist-1.png" width="768"></p>
</div>
<div id="feature-feature-correlations" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#feature-feature-correlations" class="anchor"></a>Feature-feature correlations</h2>
<p>This plot illustrates the distribution of Spearman correlation coefficients for pairs of features, calculated from the log(CPM) values obtained via the <code>cpm</code> function from <code>edgeR</code>, with a prior.count of 2. Only non-constant features are considered.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureCorrDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Feature-feature correlation"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/featureCorrSepHist-1.png" width="768"></p>
</div>
<div id="library-size-vs-fraction-zeros" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#library-size-vs-fraction-zeros" class="anchor"></a>Library size vs fraction zeros</h2>
<p>This scatter plot shows the association between the total count (column sums) and the fraction of zeros observed per sample.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">sampleDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Libsize</span>, y <span class="op">=</span> <span class="va">Fraczero</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Library size"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Fraction zeros"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/libsizeFraczeroSepScatter-1.png" width="768"></p>
</div>
<div id="mean-expression-vs-fraction-zeros" class="section level2 tabset tabset-pills">
<h2 class="hasAnchor">
<a href="#mean-expression-vs-fraction-zeros" class="anchor"></a>Mean expression vs fraction zeros</h2>
<p>This scatter plot shows the association between the average abundance and the fraction of zeros observed per feature. The abundance is defined as the log(CPM) values as calculated by <code>edgeR</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">featureDF</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">AveLogCPM</span>, y <span class="op">=</span> <span class="va">Fraczero</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.75</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, nrow <span class="op">=</span> <span class="va">colRow</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Average log CPM"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Fraction zeros"</span><span class="op">)</span> <span class="op">+</span> <span class="va">thm</span></code></pre></div>
<p><img src="comparison_reference_files/figure-html/logCPMFraczeroSepScatter-1.png" width="768"></p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-Chen2014Dispersion">
<p>Chen, Yunshun, Aaron TL Lun, and Gordon K Smyth. 2014. “Differential Expression Analysis of Complex RNA-Seq Experiments Using edgeR.” <em>In Statistical Analysis of Next Generation Sequence Data. Somnath Datta and Daniel S Nettleton (Eds), Springer, New York</em>. <a href="https://doi.org/10.1007/978-3-319-07212-8_3">https://doi.org/10.1007/978-3-319-07212-8_3</a>.</p>
</div>
<div id="ref-Love2014DESeq2">
<p>Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.” <em>Genome Biology</em> 15: 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.</p>
</div>
<div id="ref-Robinson2010edgeR">
<p>Robinson, Mark D, Davis J McCarthy, and Gordon K Smyth. 2010. “edgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.” <em>Bioinformatics</em> 26: 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616">https://doi.org/10.1093/bioinformatics/btp616</a>.</p>
</div>
<div id="ref-Robinson2010TMM">
<p>Robinson, Mark D, and Alicia Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.” <em>Genome Biology</em> 11: R25. <a href="https://doi.org/10.1186/gb-2010-11-3-r25">https://doi.org/10.1186/gb-2010-11-3-r25</a>.</p>
</div>
<div id="ref-Soneson2018">
<p>Soneson, CharlotteI, and Mark D Robinson. 2018. “Towards Unified Quality Verification of Synthetic Count Data with countsimQC.” <em>Bioinformatics</em> 34: 691–92. <a href="https://doi.org/10.1093/bioinformatics/btx631">https://doi.org/10.1093/bioinformatics/btx631</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
