<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>dyngen: A multi-modal simulator for spearheading single-cell omics analyses — dyngen • dyngen</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="dyngen: A multi-modal simulator for spearheading single-cell omics analyses — dyngen"><meta property="og:description" content="A toolkit for generating synthetic single cell data."><meta property="og:image" content="https://dyngen.dynverse.org/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-7YGMM4L0JM"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-7YGMM4L0JM');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dyngen</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/installation.html">Installation instructions</a>
    </li>
    <li>
      <a href="../articles/getting_started.html">Getting started</a>
    </li>
    <li>
      <a href="../articles/showcase_backbones.html">Showcase different backbones</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced</li>
    <li>
      <a href="../articles/advanced_topics/comparison_reference.html">Advanced: Comparison to reference dataset</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/constructing_backbone.html">Advanced: Constructing a custom backbone</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/run_dyngen_from_docker.html">Advanced: Running dyngen from a docker container</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/scalability_and_runtime.html">Advanced: On scalability and runtime</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/simulating_batch_effects.html">Advanced: Simulating batch effects</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/simulating_knockouts.html">Advanced: Simulating a knockout experiment</a>
    </li>
    <li>
      <a href="../articles/advanced_topics/tweaking_parameters.html">Advanced: Tweaking parameters</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dynverse/dyngen/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>dyngen: A multi-modal simulator for spearheading single-cell omics analyses</h1>
    <small class="dont-index">Source: <a href="https://github.com/dynverse/dyngen/blob/HEAD/R/package.R" class="external-link"><code>R/package.R</code></a></small>
    <div class="hidden name"><code>dyngen.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A toolkit for generating synthetic single cell data.</p>
    </div>


    <div id="step-initialise-dyngen-model">
    <h2>Step 1, initialise dyngen model</h2>
    

<ul><li><p><code><a href="initialise_model.html">initialise_model()</a></code>: Define and store settings for all following steps. See each of the sections below for more information.</p></li>
<li><p>Use a predefined backbone:</p><ul><li><p><code><a href="backbone_models.html">list_backbones()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_bifurcating()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_bifurcating_converging()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_bifurcating_cycle()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_bifurcating_loop()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_branching()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_binary_tree()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_consecutive_bifurcating()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_trifurcating()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_converging()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_cycle()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_cycle_simple()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_linear()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_linear_simple()</a></code></p></li>
<li><p><code><a href="backbone_models.html">backbone_disconnected()</a></code></p></li>
</ul></li>
<li><p>Create a custom backbone:</p><ul><li><p><code><a href="backbone.html">backbone()</a></code></p></li>
<li><p><code><a href="bblego.html">bblego()</a></code></p></li>
<li><p><code><a href="bblego.html">bblego_linear()</a></code></p></li>
<li><p><code><a href="bblego.html">bblego_branching()</a></code></p></li>
<li><p><code><a href="bblego.html">bblego_start()</a></code></p></li>
<li><p><code><a href="bblego.html">bblego_end()</a></code></p></li>
</ul></li>
<li><p>Visualise the backbone:</p><ul><li><p><code><a href="plot_backbone_modulenet.html">plot_backbone_modulenet()</a></code></p></li>
<li><p><code><a href="plot_backbone_statenet.html">plot_backbone_statenet()</a></code></p></li>
</ul></li>
</ul></div>
    <div id="step-generate-tf-network">
    <h2>Step 2, generate TF network</h2>
    

<ul><li><p><code><a href="generate_tf_network.html">generate_tf_network()</a></code>: Generate a transcription factor network from the backbone</p></li>
<li><p><code><a href="generate_tf_network.html">tf_network_default()</a></code>: Parameters for configuring this step</p></li>
</ul></div>
    <div id="step-add-more-genes-to-the-gene-network">
    <h2>Step 3, add more genes to the gene network</h2>
    

<ul><li><p><code><a href="generate_feature_network.html">generate_feature_network()</a></code>: Generate a target network</p></li>
<li><p><code><a href="generate_feature_network.html">feature_network_default()</a></code>: Parameters for configuring this step</p></li>
<li><p><code><a href="plot_feature_network.html">plot_feature_network()</a></code>: Visualise the gene network</p></li>
</ul></div>
    <div id="step-generate-gene-kinetics">
    <h2>Step 4, generate gene kinetics</h2>
    

<ul><li><p><code><a href="generate_kinetics.html">generate_kinetics()</a></code>: Generate the gene kinetics</p></li>
<li><p><code><a href="generate_kinetics.html">kinetics_default()</a></code>, <code><a href="generate_kinetics.html">kinetics_random_distributions()</a></code>: Parameters for configuring this step</p></li>
</ul></div>
    <div id="step-simulate-the-gold-standard">
    <h2>Step 5, simulate the gold standard</h2>
    

<ul><li><p><code><a href="generate_gold_standard.html">generate_gold_standard()</a></code>: Simulate the gold standard backbone, used for mapping to cell states afterwards</p></li>
<li><p><code><a href="generate_gold_standard.html">gold_standard_default()</a></code>: Parameters for configuring this step</p></li>
<li><p><code><a href="plot_gold_mappings.html">plot_gold_mappings()</a></code>: Visualise the mapping of the simulations to the gold standard</p></li>
<li><p><code><a href="plot_gold_simulations.html">plot_gold_simulations()</a></code>: Visualise the gold standard simulations using the dimred</p></li>
<li><p><code><a href="plot_gold_expression.html">plot_gold_expression()</a></code>: Visualise the expression of the gold standard over simulation time</p></li>
</ul></div>
    <div id="step-simulate-the-cells">
    <h2>Step 6, simulate the cells</h2>
    

<ul><li><p><code><a href="generate_cells.html">generate_cells()</a></code>: Simulate the cells based on its GRN</p></li>
<li><p><code><a href="generate_cells.html">simulation_default()</a></code>: Parameters for configuring this step</p></li>
<li><p><code><a href="generate_cells.html">simulation_type_wild_type()</a></code>, <code><a href="generate_cells.html">simulation_type_knockdown()</a></code>: Used for configuring the type of simulation</p></li>
<li><p><code><a href="kinetics_noise.html">kinetics_noise_none()</a></code>, <code><a href="kinetics_noise.html">kinetics_noise_simple()</a></code>: Different kinetics randomisers to apply to each simulation</p></li>
<li><p><code><a href="plot_simulations.html">plot_simulations()</a></code>: Visualise the simulations using the dimred</p></li>
<li><p><code><a href="plot_simulation_expression.html">plot_simulation_expression()</a></code>: Visualise the expression of the simulations over simulation time</p></li>
</ul></div>
    <div id="step-simulate-cell-and-transcripting-sampling">
    <h2>Step 7, simulate cell and transcripting sampling</h2>
    

<ul><li><p><code><a href="generate_experiment.html">generate_experiment()</a></code>: Sample cells and transcripts from experiment</p></li>
<li><p><code><a href="generate_experiment.html">list_experiment_samplers()</a></code>, <code><a href="generate_experiment.html">experiment_snapshot()</a></code>, <code><a href="generate_experiment.html">experiment_synchronised()</a></code>: Parameters for configuring this step</p></li>
<li><p><code><a href="simtime_from_backbone.html">simtime_from_backbone()</a></code>: Determine the simulation time from the backbone</p></li>
<li><p><code><a href="plot_experiment_dimred.html">plot_experiment_dimred()</a></code>: Plot a dimensionality reduction of the final dataset</p></li>
</ul></div>
    <div id="step-convert-to-dataset">
    <h2>Step 8, convert to dataset</h2>
    

<ul><li><p><code><a href="convert.html">as_dyno()</a></code>, <code><a href="convert.html">wrap_dataset()</a></code>: Convert a dyngen model to a dyno dataset</p></li>
<li><p><code><a href="convert.html">as_anndata()</a></code>: Convert a dyngen model to an anndata dataset</p></li>
<li><p><code><a href="convert.html">as_sce()</a></code>: Convert a dyngen model to a SingleCellExperiment dataset</p></li>
<li><p><code><a href="convert.html">as_seurat()</a></code>: Convert a dyngen model to a Seurat dataset</p></li>
</ul></div>
    <div id="one-shot-function">
    <h2>One-shot function</h2>
    

<ul><li><p><code><a href="generate_dataset.html">generate_dataset()</a></code>: Run through steps 2 to 8 with a single function</p></li>
<li><p><code><a href="plot_summary.html">plot_summary()</a></code>: Plot a summary of all dyngen simulation steps</p></li>
</ul></div>
    <div id="data-objects">
    <h2>Data objects</h2>
    

<ul><li><p><a href="example_model.html">example_model</a>: A (very) small toy dyngen model, used for documentation and testing purposes</p></li>
<li><p><a href="realcounts.html">realcounts</a>: A set of real single-cell expression datasets, to be used as reference datasets</p></li>
<li><p><a href="realnets.html">realnets</a>: A set of real gene regulatory networks, to be sampled in step 3</p></li>
</ul></div>
    <div id="varia-functions">
    <h2>Varia functions</h2>
    

<ul><li><p>dyngen: This help page</p></li>
<li><p><code><a href="get_timings.html">get_timings()</a></code>: Extract execution timings for each of the dyngen steps</p></li>
<li><p><code><a href="combine_models.html">combine_models()</a></code>: Combine multiple dyngen models</p></li>
<li><p><code><a href="rnorm_bounded.html">rnorm_bounded()</a></code>: A bounded version of <code><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm()</a></code></p></li>
<li><p><code><a href="runif_subrange.html">runif_subrange()</a></code>: A subrange version of <code><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif()</a></code></p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="initialise_model.html">initialise_model</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  backbone <span class="op">=</span> <span class="fu"><a href="backbone_models.html">backbone_bifurcating</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># \dontshow{</span></span></span>
<span class="r-in"><span><span class="co"># actually use a smaller example </span></span></span>
<span class="r-in"><span><span class="co"># to reduce execution time during</span></span></span>
<span class="r-in"><span><span class="co"># testing of the examples</span></span></span>
<span class="r-in"><span><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="initialise_model.html">initialise_model</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  backbone <span class="op">=</span> <span class="va">model</span><span class="op">$</span><span class="va">backbone</span>,</span></span>
<span class="r-in"><span>  num_cells <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>  num_targets <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>  num_hks <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>  gold_standard_params <span class="op">=</span> <span class="fu"><a href="generate_gold_standard.html">gold_standard_default</a></span><span class="op">(</span>census_interval <span class="op">=</span> <span class="fl">1</span>, tau <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  simulation_params <span class="op">=</span> <span class="fu"><a href="generate_cells.html">simulation_default</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    burn_time <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>    total_time <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>    census_interval <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    ssa_algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GillespieSSA2/man/ssa_etl.html" class="external-link">ssa_etl</a></span><span class="op">(</span>tau <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    experiment_params <span class="op">=</span> <span class="fu"><a href="generate_cells.html">simulation_type_wild_type</a></span><span class="op">(</span>num_simulations <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="va">model</span> <span class="op">&lt;-</span> <span class="va">model</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_tf_network.html">generate_tf_network</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_feature_network.html">generate_feature_network</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_kinetics.html">generate_kinetics</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_gold_standard.html">generate_gold_standard</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_cells.html">generate_cells</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="generate_experiment.html">generate_experiment</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Generating TF network</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sampling feature network from real network</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Generating kinetics for 32 features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Generating formulae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Generating gold standard mod changes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Precompiling reactions for gold standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running gold simulations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                  | 0 % elapsed=00s     |========                                          | 14% elapsed=00s, remaining~00s  |===============                                   | 29% elapsed=00s, remaining~00s  |======================                            | 43% elapsed=00s, remaining~00s  |=============================                     | 57% elapsed=00s, remaining~00s  |====================================              | 71% elapsed=00s, remaining~00s  |===========================================       | 86% elapsed=00s, remaining~00s  |==================================================| 100% elapsed=00s, remaining~00s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Precompiling reactions for simulations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running 1 simulations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mapping simulations to gold standard</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Simulation does not contain all gold standard edges. This simulation likely suffers from bad kinetics; choose a different seed and rerun.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing dimred</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Simulating experiment</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Certain backbone segments are not covered by any of the simulations. If this is intentional, please ignore this warning.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   Otherwise, increase the number of simulations (see `?simulation_default`) or decreasing the census interval (see `?simulation_default`).</span>
<span class="r-in"><span>  </span></span>
<span class="r-in"><span><span class="va">dataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="convert.html">wrap_dataset</a></span><span class="op">(</span><span class="va">model</span>, format <span class="op">=</span> <span class="st">"dyno"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: dynwrap</span>
<span class="r-in"><span><span class="co"># format can also be set to "sce", "seurat", "anndata" or "list"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># library(dynplot)</span></span></span>
<span class="r-in"><span><span class="co"># plot_dimred(dataset)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

